### PR TITLE
Feature/auth method insight

### PR DESCRIPTION
**Pull request by lordkyzr**
This PR addresses the output being changed from N/A to null to accurately show the state of the object as well as a bugfix for returning a null AuthPolicy object when requirement is not set to types or amount and no geolocations are set.

### CODE REVIEW

**Comment by lordkyzr** [comment]
I updated the ExampleCli output to better print the N/A or null based on your feedback.

File: src/iovation.LaunchKey.Sdk.ExampleCli/SharedServiceHelpers.cs [hunk 1]
```diff
@@ -13,17 +13,47 @@ namespace iovation.LaunchKey.Sdk.ExampleCli
...
+        {
+            return PrintNull(value.ToString());
+        }
+
+        private static String PrintNull(String value)
+        {
+            return String.IsNullOrEmpty(value) ? "null" : value;
+        }
+
+        public static String PrintNAForNull(AuthorizationResponseType? value)
```
Comment (lordkyzr) [inline src/iovation.LaunchKey.Sdk.ExampleCli/SharedServiceHelpers.cs]: Just make everything print nulls

File: src/iovation.LaunchKey.Sdk.ExampleCli/SharedServiceHelpers.cs [hunk 2]
```diff
@@ -76,43 +106,70 @@ private static int HandleWebhook(IServiceClient serviceClient)
...
 
-
         private static void PrintAuthorizationResponse(AuthorizationResponse authResponse)
         {
             Console.WriteLine($"Auth response was:");
             Console.WriteLine($"    Authorized:     {authResponse.Authorized}");
-            Console.WriteLine($"    Type:           {naForNull(authResponse.Type)}");
-            Console.WriteLine($"    Reason:         {naForNull(authResponse.Reason)}");
-            Console.WriteLine($"    Denial Reason:  {naForNull(authResponse.DenialReason)}");
+            Console.WriteLine($"    Type:           {PrintNAForNull(authResponse.Type)}");
```
Comment (bporter125) [inline src/iovation.LaunchKey.Sdk.ExampleCli/SharedServiceHelpers.cs]: I believe we decided to go back to your previous commit and just print null for all situations. @aenglander please confirm.

Comment (lordkyzr) [reply src/iovation.LaunchKey.Sdk.ExampleCli/SharedServiceHelpers.cs]: That was my understanding as well. Adam can confirm the most recent changes and resolve.

File: src/iovation.LaunchKey.Sdk.ExampleCli/SharedServiceHelpers.cs [hunk 3]
```diff
@@ -76,43 +92,70 @@ private static int HandleWebhook(IServiceClient serviceClient)
...
+                {
+                    Console.WriteLine($"       Locations: null");
+                }
+
+
             }
-            Console.WriteLine($"    Auth Methods:");
-            foreach (var item in authResponse.AuthMethods)
+
+            if(authResponse.AuthMethods == null)
```
Comment (aenglander) [inline src/iovation.LaunchKey.Sdk.ExampleCli/SharedServiceHelpers.cs]: Could be confusing if it's an empty list.

File: src/iovation.LaunchKey.Sdk/Client/BasicServiceClient.cs [hunk 4]
```diff
@@ -245,15 +245,22 @@ private AuthPolicy GetAuthPolicy(ServiceV3AuthsGetResponse authResponse)
...
-                        null,
-                        null,
-                        null,
-                        null,
-                        authPolicyLocations
-                    );
+                {
+                    if(authPolicyLocations.Count == 0)
+                    {
+                        authPolicy = null;
```
Comment (aenglander) [inline src/iovation.LaunchKey.Sdk/Client/BasicServiceClient.cs]: This seems like off logic. There was an auth policy. It's just in a weird state. I lean towards the client shouldn't make any guesses for the API and report what the API returned. If there is an `auth_policy` object in the JSON, let's return the AuthPolicy object.

Comment (lordkyzr) [reply src/iovation.LaunchKey.Sdk/Client/BasicServiceClient.cs]: This has been updated so if it receives an AuthPolicy but it's empty it returns an AuthPolicy object that is not null.

**Pull request merged**